File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int minutes)
    {
        return ExpectedMinutesInOven() - minutes;
    }

    public int PreparationTimeInMinutes(int layers) => layers * 2;

    public int ElapsedTimeInMinutes(int numberOfLayers, int minutes)
    {
        return PreparationTimeInMinutes(numberOfLayers) + minutes;
    }

}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna


let expectedMinutesInOven = 40


let remainingMinutesInOven minutes =
    expectedMinutesInOven - minutes


let minutesForOneLayer = 2


let preparationTimeInMinutes layers =
    minutesForOneLayer * layers


let elapsedTimeInMinutes layers minutes =
    preparationTimeInMinutes layers
        |> (+) minutes
